modules/javafx.controls/src/main/java/javafx/scene/control/skin/TreeTableRowSkin.java

```diff
@@ -116,43 +112,57 @@ public TreeTableRowSkin(TreeTableRow<T> control) {
     private void setupTreeTableViewListeners() {
-        ListenerHelper lh = ListenerHelper.get(this);
         TreeTableView<T> treeTableView = getSkinnable().getTreeTableView();
         if (treeTableView == null) {
-            lh.addInvalidationListener(getSkinnable().treeTableViewProperty(), (ev) -> {
+            registerInvalidationListener(getSkinnable().treeTableViewProperty(), (x) -> {
                 unregisterInvalidationListeners(getSkinnable().treeTableViewProperty());
                 setupTreeTableViewListeners();
             });
         } else {
-            lh.addChangeListener(treeTableView.treeColumnProperty(), (ev) -> {
+            registerChangeListener(treeTableView.treeColumnProperty(), (x) -> {
                 // Fix for RT-27782: Need to set isDirty to true, rather than the
                 // cheaper updateCells, as otherwise the text indentation will not
                 // be recalculated in TreeTableCellSkin.calculateIndentation()
                 isDirty = true;
-                getSkinnable().requestLayout();
+                if (getSkinnable() != null) {
+                    getSkinnable().requestLayout();
+                }
             });
 
             DoubleProperty fixedCellSizeProperty = getTreeTableView().fixedCellSizeProperty();
             if (fixedCellSizeProperty != null) {
-                lh.addChangeListener(fixedCellSizeProperty, (ev) -> {
-                    fixedCellSize = fixedCellSizeProperty.get();
-                    fixedCellSizeEnabled = fixedCellSize > 0;
+                registerChangeListener(fixedCellSizeProperty, (x) -> {
+                    updateCachedFixedSize();
                 });
-                fixedCellSize = fixedCellSizeProperty.get();
-                fixedCellSizeEnabled = fixedCellSize > 0;
+                updateCachedFixedSize();
 
                 // JDK-8144500:
                 // When in fixed cell size mode, we must listen to the width of the virtual flow, so
                 // that when it changes, we can appropriately add / remove cells that may or may not
                 // be required (because we remove all cells that are not visible).
                 VirtualFlow<TreeTableRow<T>> virtualFlow = getVirtualFlow();
                 if (virtualFlow != null) {
-                    lh.addChangeListener(getVirtualFlow().widthProperty(), (ev) -> treeTableView.requestLayout());
+                    registerChangeListener(getVirtualFlow().widthProperty(), (x) -> {
+                        if (getSkinnable() != null) {
+                            TreeTableView<T> t = getSkinnable().getTreeTableView();
+                            t.requestLayout();
+                        }
+                    });
                 }
             }
         }
     }
 
+    private void updateCachedFixedSize() {
+        if (getSkinnable() != null) {
+            TreeTableView<T> t = getSkinnable().getTreeTableView();
```

Comment:
I know this is a short method, but I would rather see a more descriptive variable name here

Reply:
I prefer not to drag long names if a very descriptive type is right there: `TreeTableView<T> t`